File: FizzBuzz.clj
(defn fizz-buzz [n]
  (or (not-empty (str 
        (if (zero? (mod n 3)) "Fizz") 
        (if (zero? (mod n 5)) "Buzz")))
      (str n)))

(println (map fizz-buzz (range 1 101)))

File: FizzBuzz.hs
-- This program is unapologetically copied from Rosetta Code.
-- http://rosettacode.org/wiki/FizzBuzz#Haskell
main = mapM_ (putStrLn . fizzbuzz) [1..100]

fizzbuzz n = 
    show n <|> [fizz| n `mod` 3 == 0] ++ 
               [buzz| n `mod` 5 == 0]

infixr 0 <|>
d <|> [] = d
_ <|> x = concat x

fizz = "Fizz"
buzz = "Buzz"